src/platform/macos.rs

```diff
@@ -499,22 +500,16 @@ impl EventLoopBuilderExtMacOS for EventLoopBuilder {
 
 /// Additional methods on [`MonitorHandle`] that are specific to MacOS.
 pub trait MonitorHandleExtMacOS {
-    /// Returns the identifier of the monitor for Cocoa.
-    fn native_id(&self) -> u32;
```

Comment:
Hmm, I believe there is value in keeping this (and `hmonitor` on Windows too).

The reason is that the `native_id` that is returned here is directly usable in FFI calls, since and will be two different values on macOS, UUID for comparisons and `CGDirectDisplayID` as the native ID.

Maybe rename `MonitorHandleProvider::native_id` to `MonitorHandleProvider::unique_id`?

Reply:
The thing is that we use it in `ffi` of some sort right now as well. Not sure how to go about it though.

Reply:
I think you should keep all the `native_id`/`hmonitor`/..., and provide a _new_ `MonitorHandleProvider::id`, which is a `u128` whose representation isn't guaranteed.